docs/setup/metal3-setup.md

```diff
@@ -0,0 +1,225 @@
+# Intro 
+
+## Metal<sup>3</sup>
+
+Metal<sup>3</sup> (Metal-Kubed) is an open source tool that provides components for bare metal host management utilizing the Kubernetes native API. (see https://metal3.io/)
+
+### Pre-requisite Dependencies
+<details>
+  <summary>Click Here for list of Ubuntu Dependencies (22.04 LTS)</summary>
+  <br>
+
+  Make sure your packages are up to date
+  ```
+  sudo apt-get update -y
+  sudo apt-get dist-upgrade -y
+  ```
+  To avoid kernal upgrade messages, pass the `DEBIAN_FRONTEND=noninteractive` flag like so:
+  ```
+  sudo DEBIAN_FRONTEND=noninteractive apt-get update -y
+  sudo DEBIAN_FRONTEND=noninteractive apt-get dist-upgrade -y
+  ```
+  Otherwise, just press enter when it asks about the kernal updates.
+  
+  Dependencies: <br>
+  Make sure to have python3-pip installed first:
+  ```
+  sudo apt install python3-pip -y
+  ```
+  ```
+  python3 -m pip install ansible
+  sudo apt install libvirt-clients -y
+  sudo apt install qemu-kvm -y
+  sudo apt install libvirt-daemon-system -y
+  sudo apt install pkg-config -y
+  sudo apt install libvirt-dev -y
+  sudo apt install mkisofs -y
+  sudo apt install qemu -y
+  sudo apt install virtinst -y
+  sudo apt install qemu-efi -y
+  sudo apt install sshpass -y
+  pip3 install libvirt-python
+  ```
+  </details>
+
+
+## Deploying the SUSE Edge Metal<sup>3</sup> Demo
+1. Begin by cloning the SUSE Edge M3 Demo repo 
+```
+git clone https://github.com/suse-edge/metal3-demo.git
+```
+2. Copy the extra_vars.yml.example to extra_vars.yml
+```
+cp extra_vars.yml.example extra_vars.yml
+```
+
+3. Follow the instructions within extra_vars.yml to configure it according to your needs. If you would like a pre-configured file, skip to step 4, otherwise, skip to step 6. 
+- Important note: There are hardcoded memory and cpu configurations within `playbooks/setup_metal3_core.yaml` and `roles/vm/defaults/main.yaml` that are fairly intensive. You can decrease these resources to fit your environment, but do note that if the resources are too little, you may run into unexpected issues.
+- The lowest resource configuration that is confirmed to work is: 6 VCPUs and 16000 vm_memory in `playbooks/setup_metal3_core.yaml` and 8000 vm_memory in `roles/vm/defaults/main.yaml`.
+
+4. Show Pre-configured extra_vars.yml.
+<details>
+  <summary>Click here for a pre-configured extra_vars.yml file</summary>
+
+```yaml
+##
+# Whether to deploy sylva-core
+#
+deploy_sylva_core: false
+sylva_core_repo_url: https://gitlab.com/codefol/sylva-core.git
+sylva_core_branch: metal3_existing_rancher
+baremetal_repo_url: https://github.com/dbw7/charts.git
```

Comment:
This is temporarily gonna point to my charts repo which contains a hard-coded value that allows ironic and mariadb to work on the metal3 demo enviroment over localhost.

Reply:
Can you perhaps instead push the values file changes to the main charts repo?  `database_clusterIP` should probably default to localhost anyway, so shall we just land that change before this one if that's the only reason for the fork?

Reply:
`database_clusterIP` going from `localhost` to `{{ .Values.global.database_clusterIP }}` was a fix/response to:  https://github.com/suse-edge/charts/pull/8
The fix I've been searching for is an elegant way to handle this within the Metal3 demo while keeping the main charts repo the same as to not break anything but I haven't found a solution that I've liked just yet

Reply:
I think you can just set the variable as an override [here](https://github.com/suse-edge/metal3-demo/blob/main/roles/baremetal/templates/parent-chart-overrides.yaml.j2#L1) which then gets consumed applying the helm chart [here](https://github.com/suse-edge/metal3-demo/blob/main/roles/baremetal/tasks/main.yml#L81) ?

I'm not entirely sure why we're running mariadb with `hostNetwork: true` tbh - perhaps as a follow-up we can address that, but for now I'm keen to replicate the configuration you've proven to work.

Reply:
I think there are two issues here:

- To access via Service/ClusterIP the mariadb pod shouldn't be using `hostNetwork: true`
- The mariadb container image is configured with a localhost `bind-address` so it's not listening on the PodIP (leading to access denied via the ClusterIP) - I have a [test build of the image in OBS](https://build.opensuse.org/package/rdiff/home:steven.hardy:branches:isv:metal3:BCI:ironic/mariadb?opackage=mariadb&oproject=isv%3Ametal3%3ABCI%3Aironic&rev=2) which may resolve that

Reply:
This was resolved via https://github.com/suse-edge/charts/pull/27 and related mariadb image change